### ISSUE TITLE
doentry.error and jpg.error

### ISSUE THREAD

**Issue by bthelin** [philipbl/Day-One-Lightroom-Plugin#24]
No problems with the Day One plugin, but as I am trying to use the Day One 2 plugin, the photos haven't been importing.  When I check the Default Journal.dayone file, it shows errors in both the entries and photos folders.  I have attached the folders with their errors for your review.

I apologize in advance if I did something wrong and this is user error.

I also really appreciate the great work you do to enable this.  It is amazing.

Brett

[Entries-Photos_Errors.zip](https://github.com/philipbl/Day-One-Lightroom-Plugin/files/135556/Entries-Photos_Errors.zip)

**Comment by d140r** [comment]
I have the same problem since my Lightroom 6.3 -> 6.4 update.
The only difference seems to be that since the 6.4 update, the two following xml lines are omitted, just before the </dict> directive:

<key>Version</key>

<string>1.0.0</string>

I tried to edit these lines manually, but to no avail.

**Comment by philipbl** [comment]
Thanks for the information! I haven't had any problems, but I am using Lightroom CC (2015.4 Release). I will take a look this weekend and see what I can find.

**Comment by bthelin** [comment]
I should mention that I'm using the same release as you, Lightroom CC (2015.4 Release).  I hope that helps as you look into it.  It could easily be something I'm doing if you aren't having problems and we are on the same release.  I'll also look and see if I have something that could be causing the behavior.

**Comment by bthelin** [comment]
I did some more testing this morning.  Keep in mind, I'm not a programmer.  I found that the plugin does export the photo and doentry files.  If I took the files that are in the Default Journal.dayone container and remove the .error suffix, the photo and entry appear to be valid photos and an entry.

I pulled all of the .error files out of the folders, removed the .error suffixes, and launched Day One 2 again.  After a short period of time, the corrected files in the folders once again were altered to have .error suffixes.

Again, I'm not a programmer, but I hope this might narrow down some of the things that might cause the behavior.  It seems like the entry and photo files have something that Day One 2 doesn't like in them and it is throwing the error.  I'll see if there are any characteristics about the photos that could be causing the plugin to be adding something that Day One 2 doesn't like.  Especially since this isn't a problem for others using 2015.4.

**Comment by karturb** [comment]
Hi! 

I have the same error, and I detected it's caused by the date in the generated doentry file. It should be "2016-02-27T13:36Z", for instance, and for some reason the format time function returns only "2016-02-27", which causes the input error. If I manually correct the date in the file prior to opening Day One, it imports flawlessly.

**Comment by bthelin** [comment]
I can confirm that in all my doentry files, the time is missing.  I thought that was odd, but didn't try to solve that one.  Unfortunately, I can't confirm that adding the time to that field allows Day One to import.  Below is the .doentry contents that I altered to include time in the format described.

Let me know if I missed something.  File name FEE317CD94474C6090C41FF5A337609E.doentry

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>Creation Date</key>
    <date>2016-02-04T18:13Z</date>
    <key>Entry Text</key>
    <string></string>
    <key>Location</key>
    <dict>
        <key>Administrative Area</key>
        <string>Anaheim</string>
        <key>Country</key>
        <string>Orange County</string>
        <key>Latitude</key>
        <real>33.809266666667</real>
        <key>Locality</key>
        <string>South Harbor Boulevard</string>
        <key>Longitude</key>
        <real>-117.91542166667</real>
        <key>Place Name</key>
        <string>1489-1575</string>
    </dict>
    <key>Starred</key>
    <true/>
    <key>Tags</key>
	<array/>
    <key>UUID</key>
    <string>FEE317CD94474C6090C41FF5A337609E</string>
</dict>
</plist>

**Comment by philipbl** [comment]
Okay, I was able to reproduce the error and fixed the problem in e4b4066647e04451993c3646dc5a01c57db4b4be. Thanks for all of your help with this! As @karturb described, it was the time and date value. Previously this code worked fine, but something must have changed with the API, causing it not to work.

Here is the code that used to work, but now it returns only the date:

``` lua
LrDate.timeToUserFormat( date, "%Y-%m-%dT%H:%M:%SZ" ) --> 2016-02-19
```

If you replace the `T` with a space, it works fine. They must have changed how they parse the time format.

``` lua
LrDate.timeToUserFormat( date, "%Y-%m-%d %H:%M:%S" ) --> 2016-02-19 11:09:04
```

Go ahead and redownload the plugin and let me know if it works for you.

**Comment by bthelin** [comment]
I can confirm that it works.  Great work.  Thank you for helping me continue journaling.  Making it easy allows me to keep writing.  I really appreciate it.
